src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -84,7 +84,7 @@ struct LegacyCallbackWrapper
                                   PairValueIndex<Value, Index> const &value,
                                   Output const &out) const
   {
-    _callback(predicate, value.index, out);
+    _callback(predicate, (int)value.index, out);
```

Comment:
This is a more precise treatment for APIv1. I also needed this due to the default distributed callback now being templated on the value, resulting in the `unsigned -> signed` loss of precision errors.

Reply:
Added a comment.

Reply:
Needed for aggregate initialization